### PR TITLE
implement defer helper method

### LINKED ISSUE #1516: A helper method for deferring an interaction
Something like `interaction.defer()` as a helper function.

### PR DESCRIPTION
**Pull request by Milo123459**
fixes #1516

### CODE REVIEW

**Comment by Milo123459** [comment]
Looks like the reason it is failing is because of the broken CI. Should be fixed by #1518

**Comment by Milo123459** [comment]
@acdenisSK what do you think? I'm not sure if this goes against serenity "design plan" or whatever but I think this is a good addition.

File: src/model/interactions/application_command.rs [hunk 1]
```diff
@@ -243,6 +243,25 @@ impl ApplicationCommandInteraction {
...
+    /// or an [`Error::Json`] if there is an error in deserializing the
+    /// API response.
+    ///
+    /// # Errors
+    ///
+    /// [`Error::Http`]: crate::error::Error::Http
+    /// [`Error::Json`]: crate::error::Error::Json
+    pub async fn defer(&self, http: impl AsRef<Http>) -> Result<()> {
+        self.create_interaction_response(http, |f| {
+            f.kind(InteractionResponseType::DeferredUpdateMessage)
```
Comment (arqunis) [inline src/model/interactions/application_command.rs]:
```suggestion
            f.kind(InteractionResponseType::DeferredChannelMessageWithSource)
```

**Review by arqunis — CHANGES_REQUESTED** [review]

File: src/model/interactions/application_command.rs (hunk 1 above)
Comment (Milo123459) [reply src/model/interactions/application_command.rs]: Oops..

**Comment by Milo123459** [comment]
Any changes needed?

**Review by arqunis — APPROVED** [review]

**Pull request merged**